### PR TITLE
Added functionality to support custom valueOption attributes

### PR DESCRIPTION
**Pull request by matwright**
A useful feature with ZF2 form elements that have valueOptions is being able to set custom attributes on each valueOption.
In the following example the custom attribute 'data-key' is set on each valueOption.
```
$select=new \Zend\Form\Element\Select('test');
$select->setValueOptions(
    [
        ['attributes'=>['data-key'=>'value'],'value'=>'myValue','label'=>'myLabel'],
        ['attributes'=>['data-key'=>'value2'],'value'=>'myValue2','label'=>'myLabel2']
    ]

    );

echo $this->formselect($select);
```
the element view helper returns the following HTML:
```
<select name="test">
<option value="myValue" data-key="value">myLabel</option>
<option value="myValue2" data-key="value2">myLabel2</option>
</select>
```
Being able to add these attributes to valueoptions is extremely useful for passing additional values for front-end processing, like adding a boolean value for an objects status, category or timestamp, etc.

DoctrineModule currently only supports label and value attributes for its ObjectSelect feature. This PR includes functionality to define custom attributes using a option_attributes option.

The attributes would be defined using a new  'option_attributes' key :
```
$os->setOptions([
            'object_manager' => $odm,
            'target_class' => '\Entity\SomeEntity',
            'property' => 'name',
            'option_attributes' => [
            [ 'data-type'=>'getType']
            ],
            'find_method' => [
                'name' => 'findAll',
                'params' => []
           ]
        ]);
```
Whereby the key option_attributes accepts an array of key/values, the keys represent valid HTML attributes(these are already filtered in ZF2 Select) they typically include  attributes such as data-, aria-, onEvent, etc. The value (getType in the above code example) is a method which will be called on the object and used for the option attribute value.

```
data-key="value"
```

### CODE REVIEW

**Comment by matwright** [comment]
@Ocramius would it be possible to get some feedback on this PR please ?

File: src/DoctrineModule/Form/Element/Proxy.php [hunk 1]
```diff
@@ -502,7 +528,19 @@ protected function loadValueOptions()
...
+                if (!is_callable(array($object, $methodName))) {
+
+                        throw new RuntimeException(
+                            sprintf('Method "%s::%s" is not callable', $this->targetClass, $methodName)
+                        );
+                }
+                    $optionAttributes[key($optionAttribute)] =(string) $object->{$methodName}();
+            }
+
+                $options[] = array('label' => $label, 'value' => $value, 'attributes'=>$optionAttributes);
```
Comment (jaapio) [inline src/DoctrineModule/Form/Element/Proxy.php]: needs less indenting

Comment (matwright) [inline src/DoctrineModule/Form/Element/Proxy.php]: @jaapio I've cleaned up with cs-fixer

**Comment by matwright** [comment]
Can this be merged ?

**Comment by gianarb** [comment]
Thanks @matwright 
Merged a12fcd0be0c2933c387985c72af6942581621023

Can you create a new paragraph here with this feature?? :+1: 
https://github.com/doctrine/DoctrineModule/blob/master/docs/form-element.md

**Comment by matwright** [comment]
Thanks @gianarb sure will add some instructions to the docs over the next few days.

**Pull request closed without merging**